piptools/utils.py

```diff
@@ -88,6 +93,22 @@ def comment(text: str) -> str:
     return click.style(text, fg="green")
 
 
+def canonicalize_ireq(ireq: InstallRequirement) -> InstallRequirement:
+    """
+    Return a copy of ireq with canonicalized extras strings
+    """
+    ireq = copy_install_requirement(
+        ireq, extras=set(map(canonicalize_name, ireq.extras))
+    )
+    if ireq.req:
+        ireq.req.extras = set(ireq.extras)
+    return ireq
+
+
+def install_req_from_line(*args: Any, **kwargs: Any) -> InstallRequirement:
+    return canonicalize_ireq(_install_req_from_line(*args, **kwargs))
```

Comment:
It might be worthwhile to normalize extras that could be passed to `pip-compile --extra` as well. What do you think @AndydeCleyre?

Reply:
I'd like to first see the current display of the extras in that case, but I haven't been able to get those extras-names to show up at all in the output.

Reply:
I'm not inclined to do it unless someone can demonstrate how or if it would make any difference.